lib/Document.ts

```diff
@@ -182,13 +185,22 @@ export class Document {
 		if (callback) {
```

Comment:
Is there a reason you're not able to make `.save()` async at the top? It'd help with readability and flow if you could just `await` and keep everything at the same scope. This could probably be simplified down to something like:

```typescript
try {
  await promise;
  this[internalProperties].storedInDynamo = true;
  
  const returnDocument = new this.model.Document(savedItem as any);
  returnDocument[internalProperties].storedInDynamo = true;
  if (callback) {
      const localCallback: CallbackType<Document, AWSError> = callback as CallbackType<Document, AWSError>;
      localCallback(null, returnDocument);
      return;
    }
  return returnDocument;
} catch (error) {
  if (callback) {
    callback(error);
    return;
  } 
  throw error;
}
```

Reply:
@heythisispaul You can't `await` unless you are within an `async` function. I forget **exactly** what the rational was behind not making these functions `async` but I think there was a purpose behind it at the time.

It is a good point tho and it exists in most Dynamoose functions.

I think it might be a good topic to bring up for v3. It'd be interesting to know how many users are actually using callbacks in Dynamoose. And would users be willing to use [`util.callbackify`](https://nodejs.org/dist/latest-v14.x/docs/api/util.html#util_util_callbackify_original) as an alternative if we drop native support for callbacks.

Reply:
> You can't await unless you are within an async function. 

Yeah, sorry I think ultimately I was asking why this whole function could/would not be declared as `async` in it's definition. 

Now that I look at this again, it would mean that this function would change the implementation slightly since `void` would no longer be a valid return value, only `Promise<void>`.

I think it's fair to consider dropping callback support as you mentioned, but yeah that's a v3 deal.